feature: add debugger info endpoint (/json) for external tool discovery...

june07 opened the pull request:
including browser DevTools similar to Node.js

### What does this PR do?

Fixes this issue https://github.com/oven-sh/bun/issues/9609 by adding a /json endpoint which lists the debugger metadata similar to Node.js

![image](https://github.com/user-attachments/assets/1911cd46-0dc4-4dd7-866d-4913b223eef9)

This is useful to allow Bun to be debugged with browser DevTools (i.e. Chrome, Edge, etc) in addition to the Bun web debugger.

Jarred-Sumner commented:
Thank you for this

We tried doing this early on, but unfortunately the V8 inspector protocol and the WebKit debugger protocol has diverged quite a bit and much of the debugger when used this way doesn't work. It needs a bunch of code to translate between the two protocols.

june07 commented:
Ah! My fault I guess for not digging deep enough... I didn't know that. Thanks for the fast reply! And best of luck with the project moving forward. Cheers!

june07 commented:
After reflecting on this, I realize I may have closed the discussion prematurely. The JSON endpoint would still be valuable for tools like [NiM](https://github.com/june07/nimv3) and potentially others that don’t have direct access to the execution context (e.g., Bun’s CLI entry-point for parsing data).

Currently, NiM supports Deno and Node.js, and adding Bun to this list would be beneficial since all these runtimes are based on the V8 engine. Given that Bun’s debugging isn’t integrated into the browser like Node.js with Chrome DevTools, Bun developers could greatly benefit from NiM’s features for debug session management and more.

Here’s a demonstration of debugging across all three runtimes, including NiM auto-connecting to the Bun debug process:

![v8Trifecta](https://github.com/user-attachments/assets/138b6d33-3ccb-4bb5-a559-e902e93ad956)

I’ve updated [this commit](https://github.com/oven-sh/bun/commit/697330e2c207d5d3b4130717b9cb3a366dda6a24) to set the `devtoolsFrontendUrl` to Bun’s WebKit URL and removed the Chrome DevTools entries.

I’d love to get your feedback.
